R/pre_commit.R

```diff
@@ -0,0 +1,217 @@
+#' @importFrom dplyr filter
+#' @export
+data_checker <- function(datafile_log = "datafiles_log.csv",
+                         ignore_file = ".gitignore") {
+  files <- NULL # assigning null variable to avoid rcmdcheck dplyr variable assignment error
+  all_ok <- TRUE
+  suffixes <- "xlsx$|ods$|dat$|csv$|tex$|pdf$|zip$|gz$|parquet$|rda$|rds$"
+  valid_statuses <- c("published", "reference", "dummy")
+
+  # Read tracking files
+  tryCatch(
+    {
+      log_files <- utils::read.csv(datafile_log, stringsAsFactors = FALSE)
+      ign_files <- utils::read.csv(ignore_file,
+        header = FALSE,
+        stringsAsFactors = FALSE, col.names = "filename"
+      )
+      ign_text <- readLines(ignore_file, warn = FALSE)
+    },
+    error = function(e) {
+      message("Error reading configuration files: ", e$message)
+      return(FALSE)
+    }
+  )
+
+  message("=== Gitignore Validation ===")
+  if (ncol(ign_files) > 1) {
+    message("ERROR: Commas detected in .gitignore")
+    all_ok <- FALSE
+  }
+
+  space_files <- ign_files$filename[grepl(" ", ign_files$filename)]
+  if (length(space_files) > 0) {
+    message(
+      "ERROR: Spaces in .gitignore entries:\n",
+      paste("-", space_files, collapse = "\n")
+    )
+    all_ok <- FALSE
+  }
+
+  if (length(ign_text) > 0) {
+    last_line <- ign_text[length(ign_text)]
+    if (!grepl("^\\s*$", last_line)) {
+      message("ERROR: .gitignore does not end with a newline character.")
+      all_ok <- FALSE
+    }
+  }
+
+  message("\n=== Data File Validation ===")
+  current_files <- data.frame(
+    files = list.files("./", recursive = TRUE, full.names = TRUE)
+  ) |>
+    dplyr::filter(
+      grepl(suffixes, files, ignore.case = TRUE),
+      !grepl("renv|datafiles_log.csv", files)
+    )
+
+  for (file in current_files$files) {
+    rel_path <- gsub("^./", "", file)
+
+    if (!rel_path %in% log_files$filename) {
+      message(
+        "Missing entry: ", rel_path, "\n",
+        "  - Add to datafiles_log.csv with status: ",
+        "published, reference, or dummy\n"
+      )
+      all_ok <- FALSE
+      next
+    }
+
+    status <- tolower(log_files$status[log_files$filename == rel_path])
+    in_ignore <- rel_path %in% ign_files$filename
+
+    if (!status %in% valid_statuses) {
+      if (!in_ignore && !grepl("unpublished", rel_path)) {
+        message(
+          "Unmanaged file: ", rel_path, "\n",
+          "  - Status: ", log_files$status[log_files$filename == rel_path], "\n",
+          "  - Valid statuses: ", paste(valid_statuses, collapse = ", "), "\n",
+          "  - Add to .gitignore if unpublished\n"
+        )
+        all_ok <- FALSE
+      } else {
+        message(
+          "Note: ", rel_path,
+          " is unpublished (ignored) but consider updating its status\n"
+        )
+      }
+    }
+  }
+
+  if (all_ok) {
+    message("\nAll data file checks passed!")
+    return(TRUE)
+  }
+
+  message("\nCommit blocked: Data validation issues detected")
```

Comment:
Missed an else statement here I think. Currently it gives this message regardless of whether the commit was blocked or not. Also it needs to actually stop the commit, so should be something like this:
`quit(save = "no", status = 1, runLast = FALSE)`